### ISSUE TITLE
 Test Canal CNI v3.25 - Test Release 2.23

### ISSUE THREAD

**Issue by mate4st** [kubermatic/kubermatic#12349]
Test the latest Canal CNI (v3.25) deployed in a new user cluster + in a user cluster that was running Canal v3.24 previously. 

Refer to the docs in https://docs.kubermatic.com/kubermatic/main/tutorials-howtos/networking/cni-cluster-network/

Perform each test for `iptables` and `ipvs` proxy mode. That means performing 4 tests:

```[tasklist]
### Tasks
- [x] v3.25 with iptables
- [x] v3.25 with IPVS
- [x] v3.24 upgrade to v3.25 with iptables
- [x] v3.24 upgrade to v3.25 with IPVS
- [x] auto update for k8s 1.26
```

## Testing Procedure - new cluster with v3.25
1. deploy a new user cluster with Canal v3.25 and iptables/IPVS proxy mode
2. verify that all pods are healthy
3. verify that pod-to-pod, pod-to-service, pod-to-nodeport connectivity works fine
4. run [conformance tests](https://sonobuoy.io/docs/v0.56.8/), e.g. `./sonobuoy run --wait=0 --dns-pod-labels "app.kubernetes.io/name=kube-dns,app=coredns"`

## Testing Procedure - v3.24 upgrade to v3.25
1. deploy a new user cluster with Canal v3.24 and iptables/IPVS proxy mode
2. wait until all pods are up
5. upgrade CNI version to v3.25
6. verify that pod-to-pod, pod-to-service, pod-to-nodeport connectivity works fine
7. run [conformance tests](https://sonobuoy.io/docs/v0.56.8/), e.g. `./sonobuoy run --wait=0 --dns-pod-labels "app.kubernetes.io/name=kube-dns,app=coredns"`

## Testing auto update for k8s 1.26
1. Create a cluster with canal version v3.22 and Kubernetes version 1.25
2. Update the cluster to Kubernetes 1.26.
3. Verify that canal automatically upgrades to v3.23 

## Acceptance Criteria
Testing procedure is successful for all described test cases.

**Comment by embik** [comment]
All v3.25 related scenarios worked fine. I'm going to include the auto-upgrade tests in my upgrade tests since you can no longer create a cluster with v3.22 on v2.23.0 betas.

**Comment by embik** [comment]
The auto-upgrade is unfortunately bugged. Here is what happened:

1. I created a Kubernetes 1.24 cluster on KKP 2.22.4 with Canal v3.22.
2. I upgraded KKP to v2.23.0-beta.2.
3. I upgraded the user cluster to Kubernetes 1.25. The PDB error starts because `policy/v1beta1.PodDisruptionBudget` was removed in Kubernetes 1.25, not 1.26. So the auto-upgrade rule should apply when upgrading 1.24 -> 1.25.
4. I then upgraded the user cluster to Kubernetes 1.26. Still no change in CNI version (I checked the ClusterSpec).
5. I _then_ upgraded the user cluster to Kubernetes 1.27. The auto-upgrade finally worked and Canal was upgraded to 3.23, but two versions too late.
